samples/client/petstore/rust/reqwest/petstore/src/apis/pet_api.rs

```diff
@@ -452,9 +370,7 @@ impl PetApi for PetApiClient {
         let content = resp.text()?;
 
         if status.is_success() {
-            let entity: Option<UploadFileSuccess> = serde_json::from_str(&content).ok();
-            let result = ResponseContent { status, content, entity };
-            Ok(result)
+            serde_json::from_str(&content).map_err(Error::from)
```

Comment:
missing `Ok()` also i tried testing locally, didn't work for me:

`java -jar ./modules/openapi-generator-cli/target/openapi-generator-cli.jar generate -g rust -i slack_web_modified.json -o slack --skip-validate-spec --remove-operation-id-prefix --library reqwest --additional-properties useSingleRequestParameter=true,supportAsync=true,supportMultipleReturns=true`

Reply:
Hi,

I generated a slack client from official Github JSON with the same parameters as yours:

```
java -jar ./modules/openapi-generator-cli/target/openapi-generator-cli.jar generate -g rust -i https://raw.githubusercontent.com/slackapi/slack-api-specs/master/web-api/slack_web_openapi_v2.json -o /tmp/slack --skip-validate-spec --remove-operation-id-prefix --library reqwest --additional-properties useSingleRequestParameter=true,supportAsync=true,supportMultipleReturns=true
```

The client compiles and works fine: can you provide some information to reproduce the problem?

PS: in issue #6650, @HenningHolmDE tested the commit with and without the new param and didn't notice a problem.

Reply:
In the code `serde_json::from_str(&content).map_err(Error::from)` you quoted, `Ok()` is not missing, since this line produces the expected return type `Result<Entity, Error>`.

Reply:
```
supportMultipleReturns=false
```

this was the only thing that worked for me

Reply:
Do you have an example OAS to reproduce your problem?

I have a compiling client (with Rust 1.44.0 and both values for `supportMultipleReturns`).

Reply:
@bcourtine I think @seunlanlege is using https://gist.githubusercontent.com/wing328/61225b4ad2d838447340a70494e09333/raw/b42fa7cf01df2a9269b910d19393a96081a246ff/slack_web_modified.json

Reply:
Hm, using that spec file also works for me (rust stable as well as nightly):
```
java -jar ./modules/openapi-generator-cli/target/openapi-generator-cli.jar generate -g rust -i https://gist.githubusercontent.com/wing328/61225b4ad2d838447340a70494e09333/raw/b42fa7cf01df2a9269b910d19393a96081a246ff/slack_web_modified.json -o slack --skip-validate-spec --remove-operation-id-prefix --library reqwest --additional-properties useSingleRequestParameter=true,supportAsync=true,supportMultipleReturns=true
cd slack
cargo check
> Finished dev [unoptimized + debuginfo] target(s) in 0.14s
```
Or is the problem related to function instead of compilation?